samples/node/count_tokens.js

```diff
@@ -0,0 +1,316 @@
+
+  const uploadResult = await fileManager.uploadFile(
+    `${mediaPath}/jetpack.jpg`,
+    { mimeType: "image/jpeg" },
+  );
+
+  const imagePart = {
+    fileData: {
+      fileUri: uploadResult.file.uri,
+      mimeType: uploadResult.file.mimeType,
+    },
+  };
+
+  const genAI = new GoogleGenerativeAI(process.env.API_KEY);
+  const model = genAI.getGenerativeModel({
+    model: "gemini-1.5-flash",
+  });
+
+  const result = await model.countTokens([
+    "Tell me about this image.",
+    imagePart,
+  ]);
+
+  console.log(result.totalTokens);
+  // [END tokens_multimodal_image_file_api]
+}
+
+async function tokensMultimodalVideoAudioFileApi() {
+  // [START tokens_multimodal_video_audio_file_api]
+  const fileManager = new GoogleAIFileManager(process.env.API_KEY);
+
+  function waitForProcessing(fileName) {
+    return new Promise(async (resolve, reject) => {
+      let file = await fileManager.getFile(fileName);
+      while (file.state === FileState.PROCESSING) {
+        process.stdout.write(".");
+        // Sleep for 10 seconds
+        await new Promise((resolve) => setTimeout(resolve, 10_000));
+        // Fetch the file from the API again
+        file = await fileManager.getFile(fileName);
+      }
+
+      if (file.state === FileState.FAILED) {
+        reject(new Error("Video processing failed."));
+      }
+      resolve();
+    });
+  }
+
+  const uploadAudioResult = await fileManager.uploadFile(
+    `${mediaPath}/samplesmall.mp3`,
+    { mimeType: "audio/mp3" },
+  );
+
+  const uploadVideoResult = await fileManager.uploadFile(
+    `${mediaPath}/Big_Buck_Bunny.mp4`,
+    { mimeType: "video/mp4" },
+  );
+
+  await Promise.all([
+    waitForProcessing(uploadAudioResult.file.name),
+    waitForProcessing(uploadVideoResult.file.name),
+  ]);
+
+  const audioPart = {
+    fileData: {
+      fileUri: uploadAudioResult.file.uri,
+      mimeType: uploadAudioResult.file.mimeType,
+    },
+  };
+
+  const videoPart = {
+    fileData: {
+      fileUri: uploadVideoResult.file.uri,
+      mimeType: uploadVideoResult.file.mimeType,
+    },
+  };
+
+  const genAI = new GoogleGenerativeAI(process.env.API_KEY);
+  const model = genAI.getGenerativeModel({
+    model: "gemini-1.5-flash",
+  });
+
+  const result = await model.countTokens([
+    "Tell me about this audio and video.",
+    audioPart,
+    videoPart,
+  ]);
+
+  console.log(result.totalTokens);
+  // [END tokens_multimodal_video_audio_file_api]
+}
+
+async function tokensCachedContent() {
+  // [START tokens_cached_content]
+  // Generate a very long string
+  let longContentString = "";
+  for (let i = 0; i < 32001; i++) {
+    longContentString += "Purple cats drink lemonade.";
+    longContentString += i % 8 === 7 ? "\n" : " ";
+  }
+
+  const cacheManager = new GoogleAICacheManager(process.env.API_KEY);
+  const cacheResult = await cacheManager.create({
+    ttlSeconds: 600,
+    model: "models/gemini-1.5-pro-001",
```

Comment:
Why don't we use 1.5 flash here?

Reply:
I'm not entirely sure why but `gemini-1.5-flash` doesn't work:

GoogleGenerativeAIFetchError: [GoogleGenerativeAI Error]: Error fetching from https://generativelanguage.googleapis.com/v1beta/cachedContents: [400 Bad Request] Model models/gemini-1.5-flash not found or does not support caching. Use ListModels to find a model that supports createCachedContent.

But `gemini-1.5-flash-001` does. I converted it to use that for now and I'm going to ask if this is intended or a bug.

Reply:
Oh, never mind. See note here. https://ai.google.dev/gemini-api/docs/caching?lang=node So flash-001 should be correct.